tests.sh

```diff
@@ -435,9 +621,87 @@ tests_wait_file_changes() {
         return 0
     done
 }
+
+# @description Sets verbosity of testcase output.
+#
+# @arg $1 int Verbosity.
+tests:set-verbose() {
+    tests_verbose=$1
+}
+
+# @description Recursively copy specified file or directory from the testcases
+# dir to the temporary test directory.
+#
+# @arg $1 filename Filename to copy.
+tests:cp() {
+    local files=($tests_case_dir/$1)
+    shift
+
+    while [ $# -gt 1 ]; do
+        files=($files $tests_case_dir/$1)
+
+        shift
+    done
+
+    tests:debug "cp -r ${files[@]} $tests_dir/$1"
+    /bin/cp -r ${files[@]} $tests_dir/$1
+}
+
+# @description Copy specified file from testcases to the temporary test
+# directory and then source it.
+#
+# @arg $1 filename Filename to copy and source.
+tests:source() {
+    tests:cp "$1"
+
+    local source_name=$(basename $1)
+    tests:debug "{BEGIN} source $source_name"
+    builtin source "$source_name"
+    tests:debug "{END} source $source_name"
```

Comment:
strance construction "{BEGIN}" "{END}", better to write more obvious debug messages like just `source $source_name` and `$source_name sourced`